rust-toolchain.toml

```diff
@@ -0,0 +1,3 @@
+[toolchain]
+channel = "1.58.1"
+components = ["cargo", "rustc", "rustfmt", "rust-analysis"]
```

Comment:
@soywod why did you add this line? Do you only want to have those components?

Reply:
Well there is not that much available https://rust-lang.github.io/rustup/concepts/components.html, do you see any other components to add?

Reply:
We can also remove this line but I do not know which component is added by default?

Reply:
hm... I think that all components are used per default and if I think about it again, then we can leave this line there, except adding the `clippy` component! Some people might use it and I think that it's pretty useful.

Reply:
I think that this is fine now